plugin.program.mceremote/mceremote.py

```diff
@@ -0,0 +1,733 @@
+        return
+
+# Build the settings list from the _buttonData
+
+    reportmappingtable = []
+
+    for button in _buttonData:
+        if len(button[3]) == 3:
+            reportmappingtable.append(button[1])
+            reportmappingtable.append(0)
+            reportmappingtable.append(0)
+            reportmappingtable.append(0)
+            for i in button[3]:
+                reportmappingtable.append(i)
+
+# Write the ReportMappingTable data
+
+    try:
+        winreg.SetValueEx(hkey, "ReportMappingTable", 0, winreg.REG_BINARY, bytes(reportmappingtable))
+    except Exception as e:
+        dialog.ok("MCERemote", "Unexpected error writing the ReportMappingTable data: " + e.args[0])
+    except:
+        dialog.ok("MCERemote", "Unexpected error writing the ReportMappingTable data")
+
+# Finished!
+
+    winreg.CloseKey(hkey)
+    dialog.ok("MCERemote", "Done! Please restart Windows to activate the changes.")
+
+
+# **********************************************************************
+# ConfigureButtonSettings
+# -----------------------
+# Open the dialog to configure the settings
+# **********************************************************************
+def ConfigureButtonSettings():
+
+# Open the settings dialog
+
+    _settings.openSettings()
+
+# Ask the user if they want to apply these settings
+
+    if xbmcgui.Dialog().yesno("MCERemote", "Do you want to apply these settings to the remote now?"):
+        ApplyCurrentSettings(False)
+
+
+# **********************************************************************
+# RestoreDefaultButtonSettings
+# ----------------------------
+# Restore the default settings
+# **********************************************************************
+def RestoreDefaultButtonSettings():
+
+# Require confirmation from the user
+
+    dialog = xbmcgui.Dialog()
+    if not dialog.yesno("MCERemote", "This will reset the remote button settings to the Media Center defaults. Do you want to proceed?"):
+        return
+
+# Restore the default settings from the info in _buttonData
+
+    for nextbutton in _buttonData:
+        _settings.setSetting(nextbutton[0], nextbutton[2])
+
+# Ask the user if they want to apply these settings
+
+    if xbmcgui.Dialog().yesno("MCERemote", "Do you want to apply these settings to the remote now?"):
+        ApplyCurrentSettings(False)
+
+
+# **********************************************************************
+# EditKeyboardDotXML
+# --------------------
+# Edit the keyboard.xml in the user's userdata\keymaps folder.
+# **********************************************************************
+def EditKeyboardDotXML():
+
+    import os
+    import subprocess
+
+    dialog = xbmcgui.Dialog()
+
+# Check if we are updating KeyMapEdit.exe
+
+    doupdate = _settings.getSetting("update_keyedit")
+    if doupdate == "true":
+        if dialog.yesno("MCERemote", "Do you want to update the keymap editor from Sourceforge?"):
```

Comment:
You can't update keymaps from remote locations.

Reply:
It does not update the keymap from a remote location. If no keyboard.xml exists in the user's userdata directory it copies a template from the resources/data folder. The only thing it downloads is my keymap editor applet. I guess I can remove the option to download the applet, but then users have to either hand edit the keymap in a text editor or manually download and copy the applet.